### PR TITLE
BE(audience):created get and post api for audience and also entity table and done migration

### PR DESCRIPTION
**Pull request by thobhanikavanm**
### description
- created audience module 
- added get post update delete api
#ticket
-https://teamselection.atlassian.net/browse/TOG-28

### CODE REVIEW

File: migrations/1671102045930-create_audience_table.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class createAudienceTable1671102045930 implements MigrationInterface {
```
Comment (jaynishb) [inline migrations/1671102045930-create_audience_table.ts]:
```suggestion
export class CreateAudienceTable1671102045930 implements MigrationInterface {
```

Always make sure class name has Capital first letter

File: src/modules/audience/audience.controller.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    })
+    async addAudience(
+        @AuthUser() user: User,
+        @Body() audience: AudienceDto,
+    ): Promise<Audience> {
+        const audienceWithCreatedBy = {
+            ...audience,
+            createdBy: user,
+        };
+        return this.audienceService.addAudience(audienceWithCreatedBy);
```
Comment (jaynishb) [inline src/modules/audience/audience.controller.ts]: We don't require to store who created this audience entity, also we don't have any `createdBy` field in `audience` table


```suggestion
        return this.audienceService.addAudience(audience);
```

File: src/modules/audience/audience.controller.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    ): Promise<AudiencePageDto> {
+        return this.audienceService.getAudience(pageOptionsDto);
+    }
+
+    @Post()
+    @Roles(RoleType.USER)
+    @HttpCode(HttpStatus.OK)
+    @ApiResponse({
+        status: HttpStatus.CREATED,
+        description: 'Created audience toggle',
```
Comment (jaynishb) [inline src/modules/audience/audience.controller.ts]:
```suggestion
        description: 'Created an audience',
```

File: migrations/1671102045930-create_audience_table.ts (hunk 1 above)
Comment (thobhanikavanm) [reply migrations/1671102045930-create_audience_table.ts]: done

File: src/modules/audience/audience.controller.ts (hunk 2 above)
Comment (thobhanikavanm) [reply src/modules/audience/audience.controller.ts]: done

File: src/modules/audience/audience.controller.ts (hunk 3 above)
Comment (thobhanikavanm) [reply src/modules/audience/audience.controller.ts]: done

**Review by jaynishb — APPROVED** [review]

**Pull request merged**